util/src/test/scala/org/broadinstitute/dsde/rawls/util/ThresholdOneForOneStrategySpec.scala

```diff
@@ -0,0 +1,405 @@
+package org.broadinstitute.dsde.rawls.util
+
+import akka.actor.{Actor, ActorRef, ActorSystem, Props, SupervisorStrategy}
+import akka.actor.SupervisorStrategy.{Directive, Restart}
+import akka.testkit.TestKit
+import org.scalatest.concurrent.{Eventually, ScalaFutures}
+import org.scalatest.mock.MockitoSugar
+import org.scalatest.{BeforeAndAfterAll, FlatSpecLike, Matchers}
+import org.scalatest.time.{Seconds, Span}
+import akka.pattern.ask
+import akka.util.Timeout
+
+import scala.concurrent.duration._
+
+class ThresholdOneForOneStrategySpec extends TestKit(ActorSystem("ThresholdOneForOneStrategySpec")) with ScalaFutures with Eventually with FlatSpecLike with MockitoSugar with Matchers with BeforeAndAfterAll {
+
+  // This configures how long the calls to `whenReady(Future)` and `eventually` will wait
+  // before giving up and failing the test.
+  // See: http://doc.scalatest.org/2.2.4/index.html#org.scalatest.concurrent.Futures
+  implicit override val patienceConfig = PatienceConfig(timeout = scaled(Span(10, Seconds)))
+
+  val alwaysRestart: PartialFunction[Throwable, Directive] = {
+    case _ => Restart
+  }
+
+  object TestException extends Exception("boom")
+
+  case class ChildLifeCheck(a: ActorRef)
+
+  class SupervisorTestActor(override val supervisorStrategy: SupervisorStrategy) extends Actor {
+    override def receive = {
+      // create new child to supervise
+      case "newchild" =>
+        val ref = context.actorOf(Props(new ChildTestActor()))
+        sender() ! ref
+
+      // is this child alive?
+      case ChildLifeCheck(child) =>
+        sender() ! context.children.toSet.contains(child)
+    }
+  }
+
+  class ChildTestActor() extends Actor {
+    var counter: Int = 0
+
+    override def receive = {
+      case "increment" =>
+        counter += 1
+        sender() ! counter
+
+      case "exception" =>
+        throw TestException
+    }
+  }
+
+  "ThresholdOneForOneStrategy" should "implement OneForOneStrategy when threshold > retries" in {
+    implicit val timeout = Timeout(5.seconds)
+
+    val varInit = 1000
+    var testVar: Int = varInit
+
+    // threshold function
+    def setVar(cause: Throwable, v: Int): Unit = {
+      assertResult(TestException)(cause)
+      testVar = v
+    }
+
+    val strategy = new ThresholdOneForOneStrategy(thresholdLimit = 3, maxNrOfRetries = Option(1))(alwaysRestart)(setVar)
+    val supervisor: ActorRef = system.actorOf(Props(new SupervisorTestActor(strategy)))
+
+    // retrieve a new supervised child
+    val childRef: ActorRef = whenReady(supervisor ? "newchild") {
+      _.asInstanceOf[ActorRef]
+    }
+
+    // start count = 0
+    // increment a few times and retrieve new counts
+
+    whenReady(childRef ? "increment") { resp =>
```

Comment:
Just a note: akka TestKit has ways of testing actors in a synchronous way, so you don't need to deal with futures in tests:

http://doc.akka.io/docs/akka/2.5.3/scala/testing.html#synchronous-testing-testactorref